### ISSUE TITLE
write_literal: invalid check on raw string

### ISSUE THREAD

**Issue by sunng87** [rust-lang/rust-clippy#5233]
I'm getting this warning on my raw string literal.  `r#"Usage: handlebars-cli template.hbs '{"json": "data"}'"#`

There is a json string within it. I guess clippy has treated `{ ... : ...}`  as an empty format string.

```
warning: literal with an empty format string
  --> src/cli.rs:15:9
   |
15 |         r#"Usage: handlebars-cli template.hbs '{"json": "data"}'"#
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::write_literal)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_literal
```

```
cargo clippy -V
clippy 0.0.212 (8fbb23f 2020-02-21)
```

**Comment by flip1995** [comment]
I assume your code looks something like this:

```rust
writeln!(buf, "{}", r#"foo {"json"}"#);
```

Why not change it to


```rust
writeln!(buf, r#"foo {{"json"}}"#);
```

**Comment by sunng87** [comment]
Yes, just realized that. Thanks!
